frontend/test/metabase/scenarios/question/new.cy.spec.js

```diff
@@ -231,16 +231,16 @@ describe("scenarios > question > new", () => {
     });
 
     it("should correctly choose between 'Object Detail' and 'Table (metabase#13717)", () => {
-      cy.request("PUT", `/api/field/${ORDERS.QUANTITY}`, {
-        semantic_type: "type/PK",
-      });
```

Comment:
Setting quantity as a primary key seems like a nonsensical test to me, but I don't want to stop testing something if this is somehow important. 

@kulyk, it looks like you touched this test somewhat recently, can you see any reason we want to keep that part of the test?

Reply:
@iethree this mimics a table with composed primary keys, ie when there are > 1 columns needed to uniquely identify a row, so I believe it should actually stay here

Reply:
@kulyk is right. This was the essential part of this reproduction/issue.

Please, see: https://github.com/metabase/metabase/issues/13717

Reply:
@iethree as a possible solution, AFAIR there's an idea to add a full-screen mode for object detail (not only a modal). Would it make sense to always open full-screen object detail for queries returning a single row without making it possible to open the modal?

Reply:
Thanks, that's helpful context, and a good test. I'll need to rework the underlying code slightly to correctly account for composite keys.

Reply:
1. I have restored the API call in this test to make it a valid test of composite keys. I have changed the test assertions and description to accurately test the current filter behavior of multiple primary keys.
2. Instead of hacking-in a new way to handle composite keys, I've essentially disabled object detail view for items with composite keys in this branch. I think this can be simply and elegantly handled with currently-planned work around object detail view: namely: making it viewable independent of whether a question has a visible primary key (i.e. just show a row). 
3. This introduces a temporary regression (no object detail view for composite keys), but I think that makes sense with the direction this feature is going (ie. object detail is always reached by "zooming in"), which is not currently technically possible, because the current zoom logic relies on primary keys.

Reply:
I've noticed one difference with Object detail window now, compared to the 42.4, for example.
If you click on a primary key in a joined question it now displays ALL fields (primary and joined). Is this a desired behavior?

Reply:
@iethree I've just realized one thing, and it was my mistake when I originally wrote the repro. If you read @flamber 's repro steps, you'll notice he says to set `ORDERS.ID` type to "No semantic type" in the Data model.

For whatever reason, I overlooked that part originally.
If you try the repro following the exact steps from the issue, you'll notice it still opens the object detail modal and the results are wrong. Please take a look at this video:
https://www.loom.com/share/579f38c4ee3b48b4b3d1f3876d9d2a36

Reply:
@nemanjaglumac took a look at your video, and it makes sense to me. The issue is "zooming" in on a single record currently requires a unique primary key. if you set quantity as the primary key (or have a composite primary key), what it currently does is show you the last instance of that key in the current dataset.

This is what i was referring to in terms of a temporary regression until we can enhance object detail "zooming" to support rows without unique primary keys.